packages/react/src/components/Authenticator/hooks/useAuthenticator/__tests__/Provider.test.tsx

```diff
@@ -0,0 +1,40 @@
+import { render, screen } from '@testing-library/react';
+import * as React from 'react';
+import { Hub } from 'aws-amplify';
+
+import { Provider } from '../';
+import * as UI from '@aws-amplify/ui';
+
+jest.mock('@aws-amplify/ui', () => ({
+  ...(jest.requireActual('@aws-amplify/ui') as {}),
+  defaultAuthHubHandler: jest.fn(),
+}));
+
+const hubListenSpy = jest.spyOn(Hub, 'listen');
+const hubHandlerSpy = jest.spyOn(UI, 'defaultAuthHubHandler');
+
+describe('Authenticator.Provider', () => {
+  it('listens to Auth Hub events on init', () => {
+    render(
+      <Provider>
+        <></>
+      </Provider>
+    );
+
+    expect(hubListenSpy).toBeCalledTimes(1);
+  });
+
+  it('responds to tokenRefresh event', () => {
+    render(
+      <Provider>
+        <></>
+      </Provider>
+    );
+
+    Hub.dispatch('auth', {
+      event: 'tokenRefresh',
+    });
+
+    expect(hubHandlerSpy).toHaveBeenCalledTimes(1);
```

Comment:
This has been failing due to 

```
    Expected number of calls: 1
    Received number of calls: 0
```

but I could verify the function does get called. Looks like jest is not correctly mocking it?